### ISSUE TITLE
`esp_audio_codec` Component Needs Examples, Not Test Code

### ISSUE THREAD

**Issue by gamename** [espressif/esp-adf#1339]
**Is your feature request related to a problem? Please describe.**
Yes, it is related to a problem.  Was trying to create a simple audio playback of an embedded m4a file.  So, I went to the component registry and downloaded the official [codec](https://components.espressif.com/components/espressif/esp_audio_codec/versions/2.0.3).

There are *no* examples for correct use.  There are only tests.  After much struggle, I did manage to get working code.  However, the test code is so tightly coupled with the application, it is extremely difficult to separate test code from application code.

It would have been much simpler if there were concrete examples instead of "just look at the tests". This was extremely frustrating and (more importantly) time consuming.

**Describe the solution you'd like**
It would be great if there were simple examples showing how to use the component. Again, *not* test code, example code.

**Describe alternatives you've considered**
I tried various other audio support facilities like minimp3, but I would rather use the "official" components.  

**Additional context**

I would like to mention that @TempoTian was extremely [helpful](https://github.com/espressif/esp-adf/issues/1338) in getting my app to work.  But again, the code is riddled with test scaffolding and therefore not very clean/concise.

**Comment by jason-mao** [comment]
@gamename Yes, I found that it is not ideal for demonstrating the codec API. We will add some examples for it.
